packages/alfa-css/src/grammars/font-weight.ts

```diff
@@ -0,0 +1,48 @@
+import * as Lang from "@siteimprove/alfa-lang";
+import { Grammar } from "@siteimprove/alfa-lang";
+import { Ident, Number, Token, TokenType } from "../alphabet";
+import { whitespace } from "../grammar";
+import { FontWeight } from "../properties/font-weight";
+
+type Production<T extends Token> = Lang.Production<Token, FontWeight, T>;
+
+const ident: Production<Ident> = {
+  token: TokenType.Ident,
+  prefix(token) {
+    switch (token.value) {
+      case "normal":
+        return { type: "absolute", value: 400 };
+      case "bold":
+        return { type: "absolute", value: 700 };
+      case "bolder":
+      case "lighter":
+        return { type: "relative", value: token.value };
+    }
+
+    return null;
+  }
+};
+
+const number: Production<Number> = {
+  token: TokenType.Number,
+  prefix(token) {
+    switch (token.value) {
+      case 100:
+      case 200:
+      case 300:
+      case 400:
+      case 500:
+      case 600:
+      case 700:
+      case 800:
+      case 900:
+        return { type: "absolute", value: token.value };
+    }
+    return { type: "absolute", value: 400 };
```

Comment:
This is not quite according to spec. The grammar looks as such:

```
<font-weight-absolute> = [normal | bold | <number>]
```

For `<number>`, it is stated that:

> Only values greater than or equal to 1, and less than or equal to 1000, are valid, and all other values are treated as parse errors.

Reply:
Not sure I agree about the definition. The spec (see below) states that the number can be any hundred from inclusive 100 to inclusive 900. In the spec linked below nothing is written about parse errors, so currently it just falls back to the "normal". Maybe it should return null?

![image](https://user-images.githubusercontent.com/770169/44167181-c5149380-a0cd-11e8-88c6-e6e788965288.png)
Source: [https://www.w3.org/TR/css-fonts-3/#font-weight-prop](CSS Fonts Module Level 3).

Reply:
Well ain't this peachy; there's nothing to disagree about, we're simply looking at different levels of the same spec 🤦‍♂️ I'm looking at https://www.w3.org/TR/css-fonts/ which currently contains a working draft of the level 4 fonts module instead of the latest level 3 fonts module. To make matters worse, https://drafts.csswg.org/css-fonts/ actually contains the latest level 3 fonts module... I give up 🙄

Irrespective of the confusion brought on by the beautifully inconsistent publishing scheme of the W3C, we have to support both grammars though. This is however going to be interesting as Can I Use does _not_ contain any data on the level 4 fonts module. MDN has _some_ data (https://developer.mozilla.org/en-US/docs/Web/CSS/font-weight#Browser_compatibility), but it's incorrect and incomplete.